extensions/smallrye-metrics/deployment/src/main/java/io/quarkus/smallrye/metrics/deployment/SmallRyeMetricsProcessor.java

```diff
@@ -91,6 +90,32 @@ void beans(BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
                 SmallRyeMetricsServlet.class));
     }
 
+    @BuildStep
+    void annotationTransformers(BuildProducer<AnnotationsTransformerBuildItem> transformers) {
+        // attach @MetricsBinding to each class that contains any metric annotations
+        Collection<DotName> metricsAnnotations = new ArrayList<>();
+        metricsAnnotations.add(DotName.createSimple(Gauge.class.getName()));
+        metricsAnnotations.add(DotName.createSimple(Counted.class.getName()));
+        metricsAnnotations.add(DotName.createSimple(Timed.class.getName()));
+        metricsAnnotations.add(DotName.createSimple(Metered.class.getName()));
+
+        transformers.produce(new AnnotationsTransformerBuildItem(ctx -> {
+            if (ctx.isClass()) {
+                // skip classes in package io.smallrye.metrics.interceptors
+                if (ctx.getTarget().asClass().name().prefix().equals(DotName.createSimple("io.smallrye.metrics.interceptors")))
+                    return;
```

Comment:
If the `DotName` are created with `createSimple()`, they might end up with a null prefix. I think you should get the name and `toString()` use `startsWith()`. I wonder if we should get the package name from one of the interceptor class instead of having it hardcoded?

Reply:
I chose hard-coding because picking one particular interceptor class might create the impression to the reader that the code is related to that particular one :) but OK, I changed it to use the literal of `MetricsInterceptor`, it will be a bit safer